From 03ec2add6361d0ff80f64991514fe84d01ed4336 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Mon, 11 Mar 2019 22:04:51 +0100
Subject: [PATCH 1/1] efi_loader: fix build error for freestanding.o

Since commit f51a226436a87 ("efi_loader: provide freestanding library") in
parallel builds errors

    lib/efi_selftest/../efi_loader/efi_freestanding.o:
    file not recognized: File truncated

occur. Obviously make cannot correctly sequence parallel builds with a
dependency like ../efi_loader/efi_freestanding.o.

Symbolic links may be ugly but this solution avoids the build error.

Fixes: f51a226436a87 ("efi_loader: provide freestanding library")
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_selftest/efi_freestanding.c | 1 +
 scripts/Makefile.lib                | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)
 create mode 120000 lib/efi_selftest/efi_freestanding.c

diff --git a/lib/efi_selftest/efi_freestanding.c b/lib/efi_selftest/efi_freestanding.c
new file mode 120000
index 00000000000..4b7edd52bd9
--- /dev/null
+++ b/lib/efi_selftest/efi_freestanding.c
@@ -0,0 +1 @@
+../efi_loader/efi_freestanding.c
\ No newline at end of file
diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index ec5c41ec561..70de9bb13a6 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -390,7 +390,7 @@ $(obj)/efi_reloc.o: $(srctree)/arch/$(ARCH)/lib/$(EFI_RELOC:.o=.c) $(recordmcoun
 	$(call cmd,force_checksrc)
 	$(call if_changed_rule,cc_o_c)
 
-$(obj)/%_efi.so: $(obj)/%.o $(obj)/efi_crt0.o $(obj)/efi_reloc.o $(obj)/../efi_loader/efi_freestanding.o
+$(obj)/%_efi.so: $(obj)/%.o $(obj)/efi_crt0.o $(obj)/efi_reloc.o $(obj)/efi_freestanding.o
 	$(call cmd,efi_ld)
 
 # ACPI
-- 
2.20.1

